test/functional/lua/vim_spec.lua

```diff
@@ -319,9 +319,43 @@ describe('lua stdlib', function()
 
       return b.x[1] == 1 and b.x[2] == 2 and b.y == 5 and count == 2
              and tostring(a) ~= tostring(b)
-    ]])
+    ]]))
+
+    assert(exec_lua([[
+      local a = {}
+      local b = vim.deepcopy(a)
+
+      local count = 0
+      for _ in pairs(b) do count = count + 1 end
+
+      return getmetatable(b) ~= vim._empty_dict_mt
+        and count == 0
+        and tostring(a) ~= tostring(b)
+    ]]))
+
+    assert(exec_lua([[
+      local a = vim.empty_dict()
+      local b = vim.deepcopy(a)
+
+      local count = 0
+      for _ in pairs(b) do count = count + 1 end
 
-    assert(is_dc)
+      return getmetatable(b) == vim._empty_dict_mt
+        and count == 0
+    ]]))
+
+    assert(exec_lua([[
+      local a = {x = vim.empty_dict(), y = {}}
+      local b = vim.deepcopy(a)
+
+      local count = 0
+      for _ in pairs(b) do count = count + 1 end
+
+      return getmetatable(b.x) == vim._empty_dict_mt
```

Comment:
A more canonical way rather than checking `vim._empty_dict_mt` manually, is to use `vim.tbl_islist(b.x)`